c++/include/orc/Common.hh

```diff
@@ -69,7 +69,7 @@ namespace orc {
     UNKNOWN_WRITER = INT32_MAX
   };
 
-  enum CompressionKind {
+  enum CompressionKind : std::int64_t {
```

Comment:
The definition of CompressionKind_MAX already makes CompressionKind a int64 type. Enum base type is a C++ 11 feature and should at least be configurable through orc-config.hh

Reply:
What is the reason for these constants?

Reply:
We can potentially get rid of this enum and use proto::CompressionKind instead. It can be done in a separate change.

Reply:
I mean that constants CompressionKind_MAX and WriterVersion_MAX are not used and can be removed.